perfaware/sim86/shared/Sim8086.cpp

```diff
@@ -0,0 +1,162 @@
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <string>
+#include <sstream>
+#include "sim86_shared.h"
+#pragma comment (lib, "sim86_shared_debug.lib")
+#define ArrayCount(Array) (sizeof(Array) / sizeof((Array)[0]))
+
+s16 Registers[15] = {};
+
+char const* RegisterNames[][3] =
+{
+    {"", "", ""},
+    {"al", "ah", "ax"},
+    {"bl", "bh", "bx"},
+    {"cl", "ch", "cx"},
+    {"dl", "dh", "dx"},
+    {"sp", "sp", "sp"},
+    {"bp", "bp", "bp"},
+    {"si", "si", "si"},
+    {"di", "di", "di"},
+    {"es", "es", "es"},
+    {"cs", "cs", "cs"},
+    {"ss", "ss", "ss"},
+    {"ds", "ds", "ds"},
+    {"ip", "ip", "ip"},
+    {"flags", "flags", "flags"}
+};
+
+void SimulateInstruction(instruction Instruction)
+{
+    const char* Op = Sim86_MnemonicFromOperationType(Instruction.Op);
+    
+    if (Instruction.Op == Op_mov)
+    {
+        instruction_operand Dest = Instruction.Operands[0];
+        instruction_operand Source = Instruction.Operands[1];
+        s32 Value; // s32 to avoid issues on the bitshift right
```

Comment:
More explicit comment here.
50fe50c4503b6fdaa37546302a5b90c48d86e4a2
Having thought about it more, the issue would be avoided by doing
`Value = (Value >> 8) & 0xFF;`

Reply:
Thanks. Given what you wrote in your comment, a better solution than using 64 bits would have been to used 32 bit *unsigned* type. This does logical right shifts (fills with 0) instead of arithmetic right shits (copies most significant bit).

Reply:
By the way some programming languages like java let the programmer explicitly decide what kind of shift to use by providing different operators, >> and >>> I believe.

Reply:
Noted! Thanks for the bit about asm instructions. 
I attempted to solve the shifting issue by using the aforementioned bitwise & in 9e26986a2b2c67060d8c6cb56372b97fc66d96fc
If you think it's appropriate, I will remove the comments. 
If you think unsigned 32 bit is better I will make the change.

Reply:
Yes using unsigned is more suited for this purpose. Its shift behavior is just naturally suited to the situation.

Reply:
Fixed a482a1ae0d45ee26f952ed35262636900fc85e0b

Reply:
Well in that case you can remove the masking operations, they're not useful anymore, right?

Reply:
Right! I was afraid cases like `mov ax, bl` would mess it up, but I checked the reference manual and that can't happen.
Fixed 11c0a6ba0e5515c67416cc7a5acae21746b240cf